DigitalLearningSolutions.Data/Helpers/ExternalApis/LearningHubApiClient.cs

```diff
@@ -0,0 +1,106 @@
+﻿namespace DigitalLearningSolutions.Data.Helpers.ExternalApis
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using DigitalLearningSolutions.Data.Models.LearningHubApiClient;
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+
+    public interface ILearningHubApiClient
+    {
+        Task<ResourceSearchResult> SearchResource(
+            string text,
+            int? offset = null,
+            int? limit = null,
+            IEnumerable<string>? resourceTypes = null
+        );
+
+        Task<ResourceReferenceWithResourceDetails> GetResourceByReferenceId(int resourceReferenceId);
+
+        Task<BulkResourceReference> GetBulkResourcesByReferenceIds(
+            IEnumerable<int> resourceReferenceIds
+        );
+    }
+
+    public class LearningHubApiClient : ILearningHubApiClient
+    {
+        private readonly HttpClient client;
+
+        public LearningHubApiClient(HttpClient httpClient, IConfiguration config)
+        {
+            string learningHubOpenApiKey = config.GetLearningHubOpenApiKey();
+            string learningHubOpenApiBaseUrl = config.GetLearningHubOpenApiBaseUrl();
+
+            client = httpClient;
+            client.BaseAddress = new Uri(learningHubOpenApiBaseUrl);
+            client.DefaultRequestHeaders.Accept.Clear();
+            client.DefaultRequestHeaders.Add("X-API-KEY", learningHubOpenApiKey);
+        }
+
+        public async Task<ResourceSearchResult> SearchResource(
+            string text,
+            int? offset = null,
+            int? limit = null,
+            IEnumerable<string>? resourceTypes = null
+        )
+        {
+            var queryString = GetSearchQueryString(text, offset, limit, resourceTypes);
+
+            var response = await client.GetStringAsync($"/Resource/Search?{queryString}");
+            var result = JsonConvert.DeserializeObject<ResourceSearchResult>(response);
+            return result;
+        }
+
+        public async Task<ResourceReferenceWithResourceDetails> GetResourceByReferenceId(int resourceReferenceId)
+        {
+            var response = await client.GetStringAsync($"/Resource/{resourceReferenceId}");
+            var result = JsonConvert.DeserializeObject<ResourceReferenceWithResourceDetails>(response);
+            return result;
+        }
+
+        public async Task<BulkResourceReference> GetBulkResourcesByReferenceIds(
+            IEnumerable<int> resourceReferenceIds

```

Comment:
Given that an empty collection of ints could be passed in, and we know that we will get empty lists backs, we can probably bypass the unnecessary call and just return the model with the empty lists (or maybe null if you think that might be better)

Reply:
We should keep the return type correct (so no null), but I'm not sure whether we should short circuit the API call or not. I kind of feel like the API client should always call the API, and perhaps any calling method should be deciding whether it should call the API or not.

Reply:
Yeah, that sounds reasonable, it can stay as it is.